src/k8s/cmd/util/hooks.go

```diff
@@ -0,0 +1,92 @@
+package cmdutil
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"syscall"
+
+	"github.com/spf13/cobra"
+)
+
+// paths to validate if root in the owner
+var pathsOwnershipCheck = []string{"/sys", "/proc", "/dev/kmsg"}
+
+// HookVerifyResources checks ownership of dirs required for k8s to run.
+// HookVerifyResources validates AppArmor configurations.
+// If potential issue found pops up warning.
+func HookVerifyResources() func(*cobra.Command, []string) {
+	return func(cmd *cobra.Command, args []string) {
+		var warnList []string
+		for _, path := range pathsOwnershipCheck {
+			if msg, err := validateRootOwnership(path); err != nil {
+				cmd.PrintErrf(err.Error())
+			} else {
+				warnList = append(warnList, msg)
+			}
+		}
+
+		if armor, err := checkAppArmor(); err != nil {
+			cmd.PrintErr(err.Error())
+		} else if len(armor) > 0 {
+			warnList = append(warnList, armor)
+		}
+
+		if len(warnList) > 0 {
+			cmd.PrintErrf("Warning: k8s may not run correctly due to reasons:\n%s"+
+				"If runnung inside LXD container refer to "+
```

Comment:
I cannot find a way to know that app is in lxd container

Reply:
I think if you do `cat /proc/1/environ` it should return something like `container=lxc`